larpandoracontent/LArTwoDReco/LArCosmicRay/ExtensionThroughShowerAlgorithm.cc

```diff
@@ -0,0 +1,143 @@
+/**
+ *  @file   larpandoracontent/LArTwoDReco/LArCosmicRay/ExtensionThroughShowerAlgorithm.cc
+ *
+ *  @brief  Implementation of the extension through shower class.
+ *
+ *  $Log: $
+ */
+#include "Pandora/AlgorithmHeaders.h"
+
+#include "larpandoracontent/LArTwoDReco/LArCosmicRay/ExtensionThroughShowerAlgorithm.h"
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+ExtensionThroughShowerAlgorithm::ExtensionThroughShowerAlgorithm() :
+    m_maxTrackDistanceToShowerBranch(5.f),
+    m_maxShowerBranchTransverseDistance(3.f),
+    m_maxShowerBranchLongitudinalDistance(20.f),
+    m_thresholdShowerClusterCount(5)    
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------    
+    
+bool ExtensionThroughShowerAlgorithm::FindBestClusterAssociation(const ClusterVector &clusterVector, const SlidingFitResultMapPair &slidingFitResultMapPair,
```

Comment:
The beginning and end of this function is essentially identical to the ExtensionPastDeltaRay version. This looks like a good candidate for a function that defines the core logic and basic tasks associated with finding a best cluster association, to be defined in a parent/base class, with a call to a pure virtual function to be implemented in the inheriting classes to handle the case-specific details found in the middle part of the loop.